### DISCUSSION TITLE
How do i handle an array ?

### DISCUSSION THREAD

**Discussion by StefanSa** [vectordotdev/vector#10726] [Q&A] [answered]
Hi there,
have this here currently:
```
if exists(.source.ip) {
 .related.ip = [.source.ip]
}
if exists(.destination.ip) {
  .related.ip = [.destination.ip]
}
```
`.related.ip`contains the array to be filled with `.source.ip`and `.destination.ip`.
 How do i do that ?

**Comment by spencergilbert** [comment, marked as answer]
If I understand correctly you'd want to use [`append`](https://vector.dev/docs/reference/vrl/functions/#append) or [`push`](https://vector.dev/docs/reference/vrl/functions/#push).

```coffee
if exists(.source.ip) {
  .related.ip = push(.related.ip, .source.ip)
}
if exists(.destination.ip) {
  .related.ip = push(.related.ip, .destination.ip)
}
```

**Reply by StefanSa** [reply]
Hi @spencergilbert yes Spencer,
Thanks for your help

i get this error, "!" nor "err" help here.
```
 .related.ip = push(.related.ip, .destination.nat.ip)
   │  ------------- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   │  │             │
   │  │             this expression is fallible
   │  │             update the expression to be infallible
   │  or change this to an infallible assignment:
   │  .related.ip, err = push(.related.ip, .destination.nat.ip)
```

with this, no error
```
if exists(.source.ip) {
  .related.ip = push([.related.ip], [.source.ip])
}
```
but wrong json value:
```
"related":{"ip":[null,["192.168.1.254"]]}
```
i have run into this problem with `append`.
Therefore this question, how to do this with the `array` correctly.

**Reply by spencergilbert** [reply]
Ah - so it appears as though your `.related.ip` field is new/empty. If you're creating it yourself on the event, perhaps create it as an empty array first?

The error you're seeing is that the `.related.ip` field isn't guaranteed to be an array but the function requires that argument to be an array.

```coffee
.related.ip = []
if exists(.source.ip) {
  .related.ip = push(.related.ip, .source.ip)
}
if exists(.destination.ip) {
  .related.ip = push(.related.ip, .destination.ip)
}
```

**Reply by StefanSa** [reply]
Perfect Spencer, thanks for the perfect support.
i would be happy if you could help me here too.
https://github.com/vectordotdev/vector/discussions/10699

Best regards from Germany

**Reply by spencergilbert** [reply]
> Perfect Spencer, thanks for the perfect support. i would be happy if you could help me here too. #10699
> 
> Best regards from Germany

Of course! I just left a comment there too - I think that one might be more involved though :)
